### PR TITLE
Add Union#nilable? macro method

### PR DESCRIPTION
**Pull request by Blacksmoke16**
Returns `true` if any type in the union is `Nil`.

### CODE REVIEW

File: spec/compiler/macro/macro_methods_spec.cr [hunk 1]
```diff
@@ -1960,6 +1960,35 @@ module Crystal
         assert_macro "x", %({{x.resolve?}}), [Crystal::Union.new(["Int32".path, "String".path] of ASTNode)] of ASTNode, "(Int32 | String)"
         assert_macro "x", %({{x.resolve?}}), [Crystal::Union.new(["Int32".path, "Unknown".path] of ASTNode)] of ASTNode, "nil"
       end
+
+      describe "#nilable?" do
```
Comment (Sija) [inline spec/compiler/macro/macro_methods_spec.cr:1964]:
```suggestion
      context "#nilable?" do
```

File: spec/compiler/macro/macro_methods_spec.cr [hunk 2]
```diff
@@ -1960,6 +1960,35 @@ module Crystal
...
+        it "without nil" do
```
Comment (Sija) [inline spec/compiler/macro/macro_methods_spec.cr:1965]: Please, use descriptive names, ditto elsewhere.

Comment (Blacksmoke16) [reply spec/compiler/macro/macro_methods_spec.cr:1965]: How could it get more descriptive?  The method is called `nilable?` and it either can or cannot have nil.

Comment (Sija) [reply spec/compiler/macro/macro_methods_spec.cr:1965]: it's suppose to be read like a sentence - see other examples.

**Comment by jhass** [comment]
Mmh, I would have expected `Union#types` to return an array of each type in the union, but it seems to return just an array with the union as its only element? If we fix this we could just do `types.includes?(Nil)` and have achieved a more general solution at the same time, no?

**Comment by Blacksmoke16** [comment]
Hold on.  I have some questions: https://play.crystal-lang.org/#/r/8xvp

The specs pass fine when the underlying type is a `TypeNode`; which also seems to be the default type of a Union.  

An actual `Union` type seems to only exist within method arguments and maybe return type.  I can use `#types` on the union and see that it has `Int32` and `Nil`, but `includes? Nil` is false.  This makes sense as to by the current code work, by first resolving the `Union` to a `TypeNode`; which means the specs are really testing `TypeNode#nilable?`.

It also seems that a `Union` within an annotation is a `Generic`?

**Comment by Blacksmoke16** [comment]
@asterite Any thoughts on what's going on there? ^

**Comment by asterite** [comment]
I didn't look into this, but the compiler will sometimes use `Union` for `A | B` and sometimes `Genreic`.

**Pull request closed without merging**